judge/models/problem.py

```diff
@@ -93,6 +94,18 @@ class SubmissionSourceAccess:
     FOLLOW = 'F'
 
 
+class VotePermission(IntEnum):
+    NONE = 0
+    VIEW = 1
+    VOTE = 2
+
+    def can_view(self):
+        return self >= VotePermission.VIEW
+
+    def can_vote(self):
+        return self >= VotePermission.VOTE
```

Comment:
If I cared I'd ask for this to be `@property`, but this has dragged on for long enough.